gnpy/core/request.py

```diff
@@ -319,8 +247,12 @@ def compute_constrained_path(network, req):
             candidate.sort(key=lambda x: sum(network.get_edge_data(x[i],x[i+1])['weight'] for i in range(len(x)-2)))
             total_path = candidate[0]
         else:
-            if req.loose_list[req.nodes_list.index(n)] == 'loose':
-                print(f'\x1b[1;33;40m'+f'Request {req.request_id} could not find a path crossing {nodes_list} in network topology'+ '\x1b[0m')
+            # TODO: better account for individual oose and strict node
+            # to ease: suppose that one strict makes the whole liste strict (except for the
+            # last node which is the transceiver)
+            if 'STRICT' not in req.loose_list[:-6]:
```

Comment:
what is the number 6 in there? if it's about `len('STRICT')`, please call it `len('STRICT')` (also on other lines)

but anyway, in this context this looks like mixing up substring search with list indexing. What is this code doing?

Reply:
Yes exactly. node_list contains the list of nodes the path must cross (user constraint). the last node is the receiver. Loose_list[i] sais if constraint node_list[i] is STRICT or LOOSE. Of course the last item of the list being the receiver, is a STRICT constraint.
This individual definition of loose and strict constraint has not been entirely coded yet because of the complexity of combinations and because nobody is asking for it. For the moment, we simply apply STRICT to all nodes if only one of the node is a strict constraint. 
at this stage of the code, it means that the constraint could not be applied. so if the string STRICT is in the loose_list (except for the destination), the constraint  STRICT is applied for all nodes in nodes list and there is not path satisfying it, else the computation is performed without any node constraint.